src/HigherOrderComponent/MapProvider/MapProvider.jsx

```diff
@@ -54,19 +54,13 @@ class MapProvider extends React.Component {
       ready: false
     };
 
-    if (props.map instanceof OlMap) {
-      this.state = {
-        map: props.map,
+    Promise.resolve(props.map).then((map) => {
```

Comment:
What happens if `props.map` is being rejected? Could you provide a test?

Reply:
Good question. On it right now.

Reply:
See https://github.com/terrestris/react-geo/pull/373/commits/8efe62e6ea201d485bf45eb866656a3a196355d3

Reply:
Awesome! :+1: